pkg/controller/plan/kubevirt.go

```diff
@@ -297,6 +297,88 @@ func (r *KubeVirt) DeleteImporterPods(pvc core.PersistentVolumeClaim) (err error
 	return
 }
 
+func (r *KubeVirt) DeleteJobs(vm *plan.VMStatus) (err error) {
+	vmLabels := r.vmAllButMigrationLabels(vm.Ref)
+	list := &batch.JobList{}
+	err = r.Destination.Client.List(
+		context.TODO(),
+		list,
+		&client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(vmLabels),
+			Namespace:     r.Plan.Spec.TargetNamespace,
+		},
+	)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	jobNames := []string{}
+	for _, job := range list.Items {
+		err = r.Destination.Client.Delete(context.TODO(), &job, &client.DeleteOptions{})
+		if err != nil {
+			err = liberr.Wrap(err)
+			r.Log.Error(
+				err,
+				"Deleting job failed.",
+				"job",
+				path.Join(
+					job.Namespace,
+					job.Name))
+			continue
+		}
+
+		r.Log.Info(
+			"Deleted job.",
+			"job",
+			path.Join(
+				job.Namespace,
+				job.Name))
+
+		jobNames = append(jobNames, job.Name)
+	}
+
+	// Remove pods created by jobs
```

Comment:
that's surprising, aren't they (the pods) set with the job as their owner?

Reply:
I may have forgotten to set PropagationPolicy

Reply:
keeping this as PropagationPolicy doesn't seem to do its thign